conference/cfp.py

```diff
@@ -0,0 +1,419 @@
+
+
+def save_information_from_speaker_form(user, cleaned_data):
+    user.first_name = cleaned_data['users_given_name']
+    user.save()
+
+    ap = user.attendeeprofile
+    ap.phone = cleaned_data['phone']
+    ap.is_minor = cleaned_data['is_minor']
+    ap.gender = cleaned_data['gender']
+    ap.job_title = cleaned_data['job_title']
+    ap.company = cleaned_data['company']
+    ap.company_homepage = cleaned_data['company_homepage']
+    ap.setBio(cleaned_data['bio'])
+    ap.save()
+
+    speaker, _ = Speaker.objects.get_or_create(user=user)
+    return speaker
+
+
+def add_speaker_to_talk(speaker, talk):
+    ts, _ = TalkSpeaker.objects.get_or_create(talk=talk, speaker=speaker)
+    return None
+
+
+class AddSpeakerToTalkForm(forms.ModelForm):
+
+    users_given_name = forms.CharField(label="Name of the speaker")
+    is_minor = forms.BooleanField(
+        label="Are you a minor?",
+        help_text=(
+            "Please select this checkbox if you're going to be under 18"
+            "years old on July 8th 2019"
+        ),
+        # required=False, because django forms... it means that unless someone
+        # is a minor we don't provide a value.
+        required=False,
+    )
+    job_title = forms.CharField(
+        label="Job title",
+        help_text="eg: student, developer, CTO, js ninja, BDFL",
+        max_length=50,
+        required=False,
+    )
+    phone = PhoneNumberField(
+        help_text=(
+            "We require a mobile phone number for all speakers "
+            "for last minute contacts and in case we need "
+            "timely clarification (if no reponse to previous emails).<br>"
+            "Use the international format, eg: +39-055-123456.<br />"
+            "This number will <strong>never</strong> be published."
+        ),
+        max_length=30,
+    )
+    gender = forms.CharField(
+        label="Type your gender",
+        max_length=32,
+        required=False,
+    )
+    company = forms.CharField(
+        label="Your company", max_length=50, required=False
+    )
+    company_homepage = forms.URLField(
+        label="Company homepage", required=False
+    )
+    bio = forms.CharField(
+        label="Compact biography",
+        help_text=(
+            "Please enter a short biography (one or two paragraphs) <br />"
+            "Do not paste your CV!"
+        ),
+        widget=forms.Textarea(),
+    )
+
+    class Meta:
+        model = AttendeeProfile
+        fields = [
+            'users_given_name',
+            'job_title',
+            'phone',
+            'bio',
+            'gender',
+            'is_minor',
+            'company',
+            'company_homepage',
+        ]
+
+
+class ProposalForm(forms.ModelForm):
+
+    type = forms.ChoiceField(
+        label='Type', required=True,
+        choices=TALK_TYPE,
+    )
+
+    # Talk tags
+    tags = TagField(required=True, widget=LabelWidget(model=ConferenceTag))
+
+    title = TalkBaseForm.base_fields["title"]
```

Comment:
I guess this is a way to compose both forms without using inheritance, right?

Reply:
yes, it's basically copy of whatever was the setup for previous CFP form, w/o too many modifications. Most of that can probably be moved either to this form or to some better setup, but we can probably think about it on the post-ep sprints. ;)